### DISCUSSION TITLE
How to declare types of {...props} spread as style props: Example Below

### DISCUSSION THREAD

**Discussion by piyush97** [chakra-ui/chakra-ui#4439] [Q&A] [answered]
So I'm passing my styles to my custom component using `{...props}` where props can generally range from `w="10"` or `display={{base: 'none', md: 'unset'}}`

So, what should be prop's type for my custom component? 

### Example Code:
```typescript

type SidebarContainerProps = {
  props?: any;
};
const SidebarContainer: React.FC<SidebarContainerProps> = (props) => {
  return (<Box {...props}></Box>)}
  
  ```

**Comment by nikolovlazar** [comment, marked as answer]
Since you're applying this to a `Box` component, you can set the type to `BoxProps`. Here's a codesandbox example: https://codesandbox.io/s/box-props-3q6cx

**Reply by nikolovlazar** [reply]
But instead of adding `props` in your type, extend your type with the `BoxProps`, like so:
```typescript
type SidebarContainerProps = BoxProps & {
  // ... more custom props here
};
```

**Reply by piyush97** [reply]
Thanks, BTW is this the best practice or there is any other `ideal` way of doing it also?

**Reply by nikolovlazar** [reply]
This is the best practice. It's how Types work in TypeScript. If you don't plan on adding more custom props, you can directly set your props to `BoxProps`. If you do, create your type, extend it with `BoxProps` and add your additional custom props.

**Reply by piyush97** [reply]
Thanks again! :)

**Reply by nikolovlazar** [reply]
You got it! 🙌
